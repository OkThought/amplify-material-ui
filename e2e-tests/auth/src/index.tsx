import React from 'react';
import ReactDOM from 'react-dom';
import Amplify from 'aws-amplify';

import App from './App';

Amplify.configure({
  Auth: {
    // REQUIRED only for Federated Authentication - Amazon Cognito Identity Pool ID
    identityPoolId: process.env.REACT_APP_IDENTITY_POOL_ID,

    // REQUIRED - Amazon Cognito Region
    region: process.env.REACT_APP_REGION,

    // OPTIONAL - Amazon Cognito User Pool ID
    userPoolId: process.env.REACT_APP_USER_POOL_ID,

    // OPTIONAL - Amazon Cognito Web Client ID (26-char alphanumeric string)
    userPoolWebClientId: process.env.REACT_APP_USER_POOL_WEB_CLIENT_ID,
  },
});

ReactDOM.render(<App />, document.getElementById('root'));
