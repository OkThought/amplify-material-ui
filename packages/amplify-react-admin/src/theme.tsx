import { createMuiTheme } from '@material-ui/core/styles';

declare module '@material-ui/core/styles/createMuiTheme' {
  interface Theme {
    drawerWidth: number;
  }
  interface ThemeOptions {
    drawerWidth: number;
  }
}

export const defaultTheme = createMuiTheme({
  drawerWidth: 240,
});
