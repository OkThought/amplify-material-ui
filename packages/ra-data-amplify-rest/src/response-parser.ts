import {
  GetListResult,
  GetOneResult,
  GetManyResult,
  GetManyReferenceResult,
  UpdateResult,
  UpdateManyResult,
  CreateResult,
  DeleteResult,
  DeleteManyResult,
  GetListParams,
  GetManyReferenceParams,
  GetOneParams,
  GetManyParams,
  UpdateParams,
  UpdateManyParams,
  CreateParams,
  DeleteParams,
  DeleteManyParams,
} from 'ra-core';

export type ResponseParserMethod<P, T> = (
  response: any,
  resource: string,
  params: P
) => T;

export interface ResponseParser {
  getList: ResponseParserMethod<GetListParams, GetListResult>;
  getOne: ResponseParserMethod<GetOneParams, GetOneResult>;
  getMany: ResponseParserMethod<GetManyParams, GetManyResult>;
  getManyReference: ResponseParserMethod<
    GetManyReferenceParams,
    GetManyReferenceResult
  >;
  update: ResponseParserMethod<UpdateParams, UpdateResult>;
  updateMany: ResponseParserMethod<UpdateManyParams, UpdateManyResult>;
  create: ResponseParserMethod<CreateParams, CreateResult>;
  delete: ResponseParserMethod<DeleteParams, DeleteResult>;
  deleteMany: ResponseParserMethod<DeleteManyParams, DeleteManyResult>;
}

export const defaultResponseParser: ResponseParser = {
  getList: (response, _resource, _params): GetListResult => {
    const { headers, data } = response;

    if (!headers || !headers.has('x-total-count')) {
      throw new Error(
        'The x-total-count header is missing in the HTTP Response. The amplify REST data provider expects responses for lists of resources to contain this header with the total number of results to build the pagination. If you are using CORS, did you declare X-Total-Count in the Access-Control-Expose-Headers header?'
      );
    }

    return {
      data,
      total: parseInt(headers.get('x-total-count').split('/').pop(), 10),
    };
  },

  getOne: (response, _resource, _params): GetOneResult => {
    return {
      data: response?.data,
    };
  },

  getMany: (response, _resource, _params): GetManyResult => {
    return {
      data: response?.data,
    };
  },

  getManyReference: (response, _resource, _params): GetManyReferenceResult => {
    const { headers } = response;

    if (!headers || !headers.has('x-total-count')) {
      throw new Error(
        'The x-total-count header is missing in the HTTP Response. The amplify REST data provider expects responses for lists of resources to contain this header with the total number of results to build the pagination. If you are using CORS, did you declare X-Total-Count in the Access-Control-Expose-Headers header?'
      );
    }

    return {
      data: response?.data,
      total: parseInt(headers.get('x-total-count').split('/').pop(), 10),
    };
  },

  update: (response, _resource, _params): UpdateResult => {
    return {
      data: response?.data,
    };
  },

  updateMany: (response, _resource, _params): UpdateManyResult => {
    return {
      data: response?.data,
    };
  },

  create: (response, _resource, params): CreateResult => {
    const { data } = response;
    return { data: { ...params.data, id: data.id } };
  },

  delete: (response, _resource, _params): DeleteResult => {
    return {
      data: response?.data,
    };
  },

  deleteMany: (response, _resource, _params): DeleteManyResult => {
    return { data: response?.data || [] };
  },
};
