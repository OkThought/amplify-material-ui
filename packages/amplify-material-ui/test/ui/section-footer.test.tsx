import * as React from 'react';
import { render } from '@testing-library/react';
import { SectionFooter } from '../../src';

describe('section-footer', () => {
  it('should be rendered correctly', () => {
    const { asFragment } = render(
      <SectionFooter>
        <div />
      </SectionFooter>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
