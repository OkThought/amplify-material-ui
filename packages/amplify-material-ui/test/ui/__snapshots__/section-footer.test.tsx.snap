// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`section-footer should be rendered correctly 1`] = `
<DocumentFragment>
  <div
    class="SectionFooter-box MuiBox-root css-g4b0on"
  >
    <div />
  </div>
</DocumentFragment>
`;
